ThePhasesOfCraftship/2_best_practice_first/testingBasics/exercises/2_Arrange_Act_Assert_Backwards/2_2_Stats_Calculator/src/index.spec.ts

```diff
@@ -1,5 +1,46 @@
+import { calculateStats } from ".";
+
 
 describe('stats calculator', () => {
 
+    describe('accepts a sequence of integers', () => {
+
+        it('returns a result if the sequence contains integers only', () => {
+            const sequence = [2, 4, 21, -8, 53, 40];
+            const result = calculateStats(sequence);
+            expect(result).toBeDefined();
+        });
+    
+        test.each([
+            [[2, "hello"], "Sequence must contain only integers."],   
+            [[2, 3.14], "Sequence must contain only integers."],      
+            [[2, NaN], "Sequence must contain only integers."],       
+            [[2, undefined], "Sequence must contain only integers."], 
+            [[2, null], "Sequence must contain only integers."]       
+        ])('throws an error when the sequence contains %p', (invalidSequence, expectedError) => {
+            expect(() => calculateStats(invalidSequence as number[])).toThrowError(expectedError);
+        });
+    });
```

Comment:
I like that you check for invalid input even though that wasn't in the requirement